submissions/5mountains/dom-api-task/js/main.js

```diff
@@ -0,0 +1,102 @@
+const styles = {
+    menuLink: 'menu__link',
+    menuLinkActive: 'menu__link--active',
+    menuItem: 'menu__item',
+    heroItem: 'hero__item',
+    heroItemActive: 'hero__item--active',
+    heroImg: 'hero__image',
+    itemIcon: 'item__icon',
+    itemTitle: 'item__caption',
+    itemPrice: 'item__price',
+    itemInfo: 'item__description'
+};
+const sectionMenu = document.querySelector('.nav__menu');
+const sectionHero = document.querySelector('.hero');
+const url = './data/data.json';
+let selectMenuItem;
+let selectHeroItem;
+
+async function fetchAsync () {
+    const response = await fetch(url);
+    const data = await response.json();
+    return data;
+}   
+  
+fetchAsync()
+    .then(data => {
+        const details = createHtml(data);
+        drawPage(details);
+        toggleTabs();
+    })
+    .catch(reason => console.log(reason.message));
+
+function createHtml(data) {
+    let navContent  = '';
+    let heroContent  = '';
+    data.pancakes.forEach((elem, index) => {
+        const menuItems = generateNavContent(elem, index); 
+        const heroItems = generateHeroContent(elem, index); 
+        navContent += menuItems;
+        heroContent += heroItems;
+    });
+    return {
+        navContent,
+        heroContent
+    };
+}
+
+function generateNavContent(elem, index) {
+    const navLink = `<a class=${styles.menuLink} href='#' id=link${index+1}>${elem.title}</a>`;
+    const navElem = `<li class=${styles.menuItem}>${navLink}</li>`;
+    return navElem;
+}
+
+function generateHeroContent(elem, index) {
+    const heroPrice = `<div class=${styles.itemPrice}>Price for portion ${elem.price} $</div>`;
```

Comment:
The same as in the previous comment.

Reply:
But in this case - it will be a really large line of the template. Or is it okay?)

Reply:
Yes, it's ok. Don't forget that you can use the Enter key for moving to another line and just write HTML there

Reply:
unfortunately, I cannot! Because in my case, when I check the parent's children in my functions, "Enter" shows as an empty string, and break program logic.

Reply:
Hm... It's weird. Could you please provide some screenshot or the link to some sandbox with your code?

Reply:
without Enters - https://user-images.githubusercontent.com/29441499/97614051-22ccf600-1a22-11eb-8f1f-86c363e2e997.jpg

with Enters - https://user-images.githubusercontent.com/29441499/97614046-206a9c00-1a22-11eb-98eb-a049dc76b494.jpg

and space or tab key also have influence...

Reply:
> Because in my case, when I check the parent's children in my functions,

Actually this is a good example, why you should not rely on DOM structure and check parents or children of some element.

In one day DOM structure can be changed - you will have additional button or element will be moved elsewhere in general. Instead of `childNodes` use `querySelector` and instead of parents use [`closest`](https://developer.mozilla.org/en-US/docs/Web/API/Element/closest)

Reply:
>  use `querySelector` 

I used querySelectorAll, I found solution with this approach, but then I had to found different and I came up with this example of code...

Reply:
>  this example of code...

Can you be more specific? :)

Reply:
In this case, I use path > parent.firstChild.nextSibling in order  to dig to the proper element end make animation. You could check it in 92 lines of this JS code. https://user-images.githubusercontent.com/29441499/97616671-8147a380-1a25-11eb-8485-5475ecf791ba.jpg

Reply:
It is a long road. First, I found the clicked menu item through the parent, then I found the hero's item according to the clicked menu item through comparison/replacement of the ID part, and finally by this hero's item - I go down via children to the item that should be animated ...

Reply:
@5Mountains , maybe @zonzujiro means that you can use `parent.querySelector('.hero__image')` insead of `parent.firstChild.nextSibling`. In this case you'll not depend on children and other DOM nodes.

Reply:
@MaxKalashnyk OMG))) Great Option! I did not know about it... I lack of experience yet...